napari/layers/layergroup/layergroup.py

```diff
@@ -0,0 +1,183 @@
+        return self._children.__getitem__(idx)
+
+    # We need more & better list handling methods
+    # moving a layer from one position to another
+    # moving a layer up/down heirarchy branches when moving its position
+    # removing a layer from a group (should remove the parent attribute)
+    def append(self, item):
+        item._parent = self
+        # FIXME - add a check for unique layergroup names in the tree
+        # FIXME - update ndim property on layergroup with self._get_ndim()
+        return self._children.append(item)
+
+    def __len__(self):
+        """Length of top-level list of children on the layergroup."""
+        return len(self._children)
+
+    @property
+    def nlayers(self):
+        return sum([1 for _ in self])
+
+    @nlayers.setter
+    def nlayers(self, value):
+        raise NotImplementedError  # user should not be able to overwrite this
+
+    def _get_extent(self):
+        """Combined extent bounding all the individual layergroup layers.
+
+        Returns
+        -------
+        tuple
+            Extent returned as tuple, ndim x 2 (min, max)
+        """
+        return combine_extents([c._get_extent() for c in self._children])
+
+    def _get_ndim(self):
+        return max([c._get_ndim() for c in self._children])
+
+    def _get_state(self):
+        """LayerGroup state as a list of state dictionaries.
+
+        Returns
+        -------
+        state : list
+            List of layer state dictionaries.
+        """
+        state = []
+        state.append(self._get_base_state())
+        for child in self:
+            if child._children is not None:
+                state.append(child._get_state())
+            else:
+                state.append(child._get_state())
+        return state
+
+    def _get_value(self):
+        """Returns nested list of all layer values in the layergroup
+        for a given mouse position and set of indices.
+
+        Returns
+        ----------
+        value : list
+            Nested list containing Values of the layer data at the coord.
+        """
+        return [c._get_value() for c in self._children]
+
+    def _set_view_slice(self):
+        """Set the view for each layer given the indices to slice with."""
+        for child in self:
+            child._set_view_slice()
+
+    def _set_highlight(self):
+        """Render layer hightlights when appropriate."""
+        for child in self:
+            child._set_highlight()
+
+    def _update_thumbnail(self, *args, **kwargs):
+        # FIXME
+        # we should do something here, leave it for later
+        pass
+
+    def refresh(self, event=None):
+        """Refresh all layer data if visible."""
+        if self.visible:
+            for child in self._children:
+                child.refresh()
+
+    @property
+    def data(self):
+        return None
+
+    @data.setter
+    def data(self, value):
+        raise NotImplementedError  # user should not be able to overwrite this
```

Comment:
I think you can just remove this.  But default `@properties` are not settable... and if you try to set `layer_group.data = ...` you'll get `AttributeError: can't set attribute`

Reply:
> Note: I notice that the base class has the setter as an abstractmethod... but in this case, I don't think it's actually working as intended. (That is, python doesn't complain at all about instantiating a subclass that doesn't define `data.setter` ... probably something to do with the @Property)

Huh, you're right about that! I didn't try it without just the setter.

Note: I'm keeping the blending mode setter on the layergroup for now (because that does not raise NotImplementedError from the base class), but this is another code smell.